Fix: Account Upgrade Reminders with simultaneous profile and account upgrades

appable0 opened the pull request:
## What
Fix issue with the Account Upgrade Reminder feature when profile and account upgrades were active at the same time. Now stores account and profile upgrades separately. Also, generally refactored the module.

Resolves [issue on Discord here](https://discord.com/channels/997079228510117908/1247241068404277369/1247241068404277369).

## Changelog Fixes
+ Fixed Account Upgrade Reminder feature with simultaneous account and profile upgrades. - appable

appable0 committed:
add/fix profile upgrades

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/fame/UpgradeReminder.kt:
```diff
@@ -0,0 +1,193 @@
+    @SubscribeEvent
+    fun onConfigLoad(event: ConfigLoadEvent) {
+        val playerStorage = ProfileStorageData.playerSpecific ?: return
+        val name = playerStorage.currentAccountUpgrade ?: return
```
We should use the `config migration fix event` instead. Then, we can remove the values `currentAccountUpgrade` and `nextAccountUpgradeCompletionTime` entirely

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/fame/UpgradeReminder.kt:
```diff
@@ -0,0 +1,193 @@
+                return when {
+                    accountUpgradePattern.anyMatches(lore) -> ACCOUNT
+                    profileUpgradePattern.anyMatches(lore) -> PROFILE
+                    else -> return null
```
return here can go

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/fame/UpgradeReminder.kt:
```diff
@@ -0,0 +1,193 @@
+    )
+    private val upgradeDurationPattern by patternGroup.pattern(
+        "duration",
+        "§8Duration: (?<duration>\\d{1,3})d"
```
is the format always in days? when not, will detection break? cant we use TimeUtils to dynamically parse the time unit?

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/config/storage/PlayerSpecificStorage.java:
```diff
@@ -31,6 +32,9 @@ public class PlayerSpecificStorage {
     public long nextAccountUpgradeCompletionTime = -1L;
 
+    @Expose
+    public UpgradeReminder.CommunityShopUpgrade communityShopUpgrade = null;
```
please include the word "account" to prevent confusion

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/config/storage/ProfileSpecificStorage.java:
```diff
@@ -613,4 +613,7 @@ public static class DianaStorage {
     public Map<SkillType, SkillAPI.SkillInfo> skillData = new HashMap<>();
+
+    @Expose
+    public UpgradeReminder.CommunityShopUpgrade communityShopUpgrade = null;
```
please include the word "profile" to prevent confusion

hannibal002 reviewed (changes requested):
not tested in game.

appable0 committed:
address requested changes

appable0 committed:
remove migrated values

appable0 commented:
Made some changes to address feedback. Two notes:
- The migration creates the CommunityShopUpgrade object with a null name if there was no active upgrade before. I added a workaround to not send reminders about upgrades with null names, but I'd love to know if there's a cleaner way to do this.
- It's always either days or the string "Instant!" currently. I switched to TimeUtils, but with an additional guard for "Instant!" so that instant upgrades remain ignored.

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/appable0/fix-profile-upgrades

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/SkyHanniMod.kt

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/appable0/fix-profile-upgrades

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/SkyHanniMod.kt

CalMWolfs committed:
fix merge

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/appable0/fix-profile-upgrades

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/SkyHanniMod.kt
#	src/main/java/at/hannibal2/skyhanni/features/fame/AccountUpgradeReminder.kt

hannibal002 committed:
Merge branch 'refs/heads/beta' into fork/fix-profile-upgrades

hannibal002 committed:
bumped config version
